crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -844,23 +899,33 @@ impl<'src> Parser<'src> {
     ///
     /// If the parser isn't positioned at a `String` token.
     fn parse_string(&mut self) -> StringType {
-        let (Tok::String { value, kind }, tok_range) = self.bump(TokenKind::String) else {
+        let (Tok::String { value, kind }, range) = self.bump(TokenKind::String) else {
             unreachable!()
         };
 
-        match parse_string_literal(value, kind, tok_range) {
+        match parse_string_literal(value, kind, range) {
             Ok(string) => string,
             Err(error) => {
-                let string = StringType::Invalid(ast::StringLiteral {
-                    // TODO(dhruvmanila): Do we need to get the source text? In the future, we
-                    // can just set a flag and the linter can query the source code if needed.
-                    value: self.src_text(tok_range).to_string().into_boxed_str(),
-                    range: tok_range,
-                    flags: kind.into(),
-                });
                 let location = error.location();
                 self.add_error(ParseErrorType::Lexical(error.into_error()), location);
-                string
+
+                if kind.is_byte_string() {
+                    StringType::Bytes(ast::BytesLiteral {
+                        value: self
+                            .src_text(range)
+                            .to_string()
+                            .into_boxed_str()
+                            .into_boxed_bytes(),
+                        range,
+                        flags: ast::BytesLiteralFlags::from(kind).with_invalid(),
+                    })
```

Comment:
Somewhat torn on whether we should keep the raw source text in the node as it's not a valid information and any downstream tool can get the same through the range.